test/auro-input.test.js

```diff
@@ -14,6 +14,23 @@ describe('auro-input', () => {
     expect(input.value).to.equal('other value');
   });
 
+  it('Sets custom pattern and setCustomValidity message', async () => {
+    const el = await fixture(html`
+      <auro-input value="other value" pattern="[0-9]+" setCustomValidity="that's not a phone number" keyValidate>
+        <span slot="label">First name:</span>
+      </auro-input>
+    `);
+    const validateSpy = sinon.spy(el, 'validate');
+    const input = el.shadowRoot.querySelector('input');
+    const text = el.shadowRoot.querySelector('.inputElement-helpText').innerHTML;
+
+    input.focus();
+    setInputValue(el, 'whatever@alaskaair.com');
+    input.blur();
+    expect(el.isValid).to.not.be.true;
+    expect(text).to.not.contain(`not a phone number`);
```

Comment:
Can we add a similar test for the real-time validation functionality? i.e. type in the input and check the validity before blurring.

Reply:
I don't think it's a bad test to add, but we are already at Code coverage: 93.57 %, so do you feel it's absolutely necessary?

Reply:
It's not so much a code coverage thing as it is making sure every new feature has a test 🤷‍♂️

Reply:
Test added.